### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by enolfc**
Refactor code to make it py2 & py3 compatible and make it work in CentOS 6, 7, and 8

### CODE REVIEW

File: glite-info-update-endpoints.spec [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
-Source:		%{name}-%{version}.src.tgz
-BuildArch:	noarch
-BuildRoot:	%{_tmppath}/%{name}-%{version}-build
+%if 0%{?el8}
+%global __python %{__python3}
+%global python_version_prefix 3
+%endif
+
+Name:             glite-info-update-endpoints
+Version:          3.0.1
```
Comment (andrea-manzi) [inline glite-info-update-endpoints.spec:7]: maybe better to move to a new minor? like 3.1.0?

Comment (andrea-manzi) [reply glite-info-update-endpoints.spec:7]: or at least should be aligned to the version bump . 3.02

Comment (enolfc) [reply glite-info-update-endpoints.spec:7]: yup, 3.0.2 at least

Comment (andrea-manzi) [inline glite-info-update-endpoints.spec:7]:
```suggestion
Version:          3.0.2
```

**Comment by gwarf** [comment]
Only looked at the PR description, but this looks great!
I would propose to tag and build a release now that #26 has been merged. And then you could rebase and we merge and use this as the new base.

**Comment by enolfc** [comment]
PR rebased

**Comment by enolfc** [comment]
@gwarf , @andrea-manzi can you review?

**Comment by enolfc** [comment]
> Only looked at the PR description, but this looks great!
> I would propose to tag and build a release now that #26 has been merged. And then you could rebase and we merge and use this as the new base.

Ah, ok, no need to re-review then before having the release done... so I'll wait :)

**Comment by gwarf** [comment]
HI @enolfc, minding having a look as this one? Once conflicts will have been resolved will it be ready for merging?

**Comment by enolfc** [comment]
I have rebased it again, should be ready for merging

**Review by gwarf — APPROVED** [review]
Cool!

LGTM.

**Comment by gwarf** [comment]
@enolfc seems there is a need to update the spec file: https://travis-ci.org/github/EGI-Foundation/glite-info-update-endpoints/jobs/742403797#L693

File: glite-info-update-endpoints.spec [hunk 2]
```diff
@@ -32,20 +46,25 @@ rm -rf %{buildroot}
...
 /var/cache/glite/glite-info-update-endpoints
+%{python_sitelib}/glite_info_update_endpoints/
+%{python_sitelib}/glite_info_update_endpoints-*.egg-info/
 %doc /usr/share/doc/glite-info-update-endpoints/README.md
 %doc /usr/share/doc/glite-info-update-endpoints/AUTHORS
 %doc /usr/share/doc/glite-info-update-endpoints/COPYRIGHT
 %doc /usr/share/doc/glite-info-update-endpoints/LICENSE.txt
 
 %changelog
+* Mon Jul 13 2020 Enol Fernandez <enol.fernandez@egi.eu> - 3.0.2-1
```
Comment (gwarf) [inline glite-info-update-endpoints.spec:65]: Please fix date.

**Review by gwarf — CHANGES_REQUESTED** [review]
Please fix date in the changelog

**Review by gwarf — APPROVED** [review]

**Pull request merged**